compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -282,6 +286,17 @@ impl Loop {
         Self { header, back_edge_start, blocks }
     }
 
+    /// Handle negative values represented as fields.
+    fn get_const_as_i128(function: &Function, id: ValueId) -> Option<i128> {
+        let (value, typ) = function.dfg.get_numeric_constant_with_type(id)?;
+        if matches!(typ, NumericType::NativeField) {
+            // Shouldn't happen, as fields don't have meaningful ordering.
```

Comment:
This comment in isolation I think adds confusion as the title and function logic alone indicates we are just attempting to fetch an i128. If we are assuming this function will only ever be called on the jump value of the loop let's comment that on the function itself and replace this with an assertion.

Reply:
Removed this in favour of an `IntegerConstant` type, which has a constructor returning `None` if it was called with a `NativeField`, so there is no confusion.